io-sim/io-sim.cabal

```diff
@@ -28,17 +28,6 @@ library
   exposed-modules:     Control.Monad.IOSim
   other-modules:       Control.Monad.IOSim.Internal
   default-language:    Haskell2010
-  other-extensions:    BangPatterns,
-                       ExistentialQuantification,
-                       FlexibleInstances,
-                       GADTSyntax,
-                       GeneralizedNewtypeDeriving,
-                       MultiParamTypeClasses,
-                       NamedFieldPuns,
-                       RankNTypes,
-                       ScopedTypeVariables,
-                       TypeApplications,
-                       TypeFamilies
```

Comment:
This is a misunderstanding of the field in the .cabal file. The `other-extensions` field says that modules in this package use these language extensions, so you have to use a compiler that supports them all. They're *dependencies* on language extensions.

It is the `default-extensions` field that *applies* the extensions to all the modules in the package. This is the one you were thinking of, and indeed it is standard practice not to use that and to list them on a per-module basis instead.

So we should drop this patch, or better: review which extensions we're using and make sure this list is accurate.

Reply:
I updated the list: `TypeFamilies` are not used (at least in the `library`) and we use `CPP` support `base >= 4.13.0`.